runtime/rococo/src/bridge_messages.rs

```diff
@@ -0,0 +1,410 @@
+		*lane == [0, 0, 0, 0]
+	}
+
+	fn maximal_pending_messages_at_outbound_lane() -> MessageNonce {
+		MAXIMAL_PENDING_MESSAGES_AT_OUTBOUND_LANE
+	}
+
+	fn estimate_delivery_confirmation_transaction() -> MessageTransaction<Weight> {
+		let inbound_data_size = InboundLaneData::<crate::AccountId>::encoded_size_hint(
+			MAXIMAL_ENCODED_ACCOUNT_ID_SIZE,
+			1,
+			1,
+		)
+		.unwrap_or(u32::MAX);
+
+		MessageTransaction {
+			dispatch_weight: MAX_SINGLE_MESSAGE_DELIVERY_CONFIRMATION_TX_WEIGHT,
+			size: inbound_data_size
+				.saturating_add(EXTRA_STORAGE_PROOF_SIZE)
+				.saturating_add(TX_EXTRA_BYTES),
+		}
+	}
+
+	fn transaction_payment(transaction: MessageTransaction<Weight>) -> crate::Balance {
+		// current fee multiplier is used here
+		bridge_runtime_common::messages::transaction_payment(
+			crate::BlockWeights::get().get(frame_support::weights::DispatchClass::Normal).base_extrinsic,
+			crate::TransactionByteFee::get(),
+			pallet_transaction_payment::Pallet::<crate::Runtime>::next_fee_multiplier(),
+			|weight| crate::constants::fee::WeightToFee::calc(&weight),
+			transaction,
+		)
+	}
+}
+
+impl<B, GI> BridgedChainWithMessages for RococoLikeChain<B, GI> {
+	fn maximal_extrinsic_size() -> u32 {
+		max_extrinsic_size()
+	}
+
+	fn message_weight_limits(_message_payload: &[u8]) -> RangeInclusive<Weight> {
+		// we don't want to relay too large messages + keep reserve for future upgrades
+		let upper_limit = messages_target::maximal_incoming_message_dispatch_weight(max_extrinsic_weight());
+
+		// we're charging for payload bytes in `With(Wococo | Rococo)MessageBridge::transaction_payment` function
+		//
+		// this bridge may be used to deliver all kind of messages, so we're not making any assumptions about
+		// minimal dispatch weight here
+
+		0..=upper_limit
+	}
+
+	fn estimate_delivery_transaction(
+		message_payload: &[u8],
+		include_pay_dispatch_fee_cost: bool,
+		message_dispatch_weight: Weight,
+	) -> MessageTransaction<Weight> {
+		let message_payload_len = u32::try_from(message_payload.len()).unwrap_or(u32::MAX);
+		let extra_bytes_in_payload = Weight::from(message_payload_len)
+			.saturating_sub(pallet_bridge_messages::EXPECTED_DEFAULT_MESSAGE_LENGTH.into());
+
+		MessageTransaction {
+			dispatch_weight: extra_bytes_in_payload
+				.saturating_mul(ADDITIONAL_MESSAGE_BYTE_DELIVERY_WEIGHT)
+				.saturating_add(DEFAULT_MESSAGE_DELIVERY_TX_WEIGHT)
+				.saturating_sub(if include_pay_dispatch_fee_cost {
+					0
+				} else {
+					PAY_INBOUND_DISPATCH_FEE_WEIGHT
+				})
+				.saturating_add(message_dispatch_weight),
+			size: message_payload_len
+				.saturating_add(EXTRA_STORAGE_PROOF_SIZE)
+				.saturating_add(TX_EXTRA_BYTES),
+		}
+	}
+
+	fn transaction_payment(transaction: MessageTransaction<Weight>) -> crate::Balance {
+		// current fee multiplier is used here
+		bridge_runtime_common::messages::transaction_payment(
+			crate::BlockWeights::get().get(frame_support::weights::DispatchClass::Normal).base_extrinsic,
```

Comment:
Should we enable a `CallFilter` for `DispatchClass::Normal` maybe?

Reply:
Do you mean because here I'm getting base weight of `frame_support::weights::DispatchClass::Normal` tx? But this function is used to 'compute' the cost of delivery/confirmation transactions, which are `normal`. I can add any filter, though if you think we'll need it. I have no clear understanding of what we're going to achieve with this bridge, so I'm open for any suggestions about the set of filters/parameters/... in this PR.

Reply:
Ah, fair enough. I thought this should depend on the type of call we will dispatch across the bridge, but it doesn't matter - thanks for explanations. 
I don't think we need to add any filters currently.